components/script/dom/htmlformelement.rs

```diff
@@ -88,6 +88,7 @@ use crate::task_source::TaskSource;
 #[derive(Clone, Copy, JSTraceable, MallocSizeOf, PartialEq)]
 pub struct GenerationId(u32);
 
+#[allow(clippy::type_complexity)]
```

Comment:
Does this need to be at the level of the whole `struct`? Can we not have allow it just for `past_names_map`?

Reply:
Of course, I can do that. I wasn't sure if it'd make for cleaner code.

Reply:
It is better to have it closer to the source of the warning, so we don't inadvertently allow new members of this struct with complex types.

Reply:
About the "cleaner code", I assume here you mean the visual noise from the `allow` directive. I guess we can just add blank lines before and after the field ([example](https://github.com/servo/servo/blob/main/components/script/dom/htmlformelement.rs#L105)) so it doesn't seem too "crammed".

Reply:
okay great! I'll add the blank lines :)

Reply:
but won't `./mach test-tidy` report errors after doing that?

Reply:
I tried it and it didn't.